files/nrpe/check_object.py

```diff
@@ -50,9 +69,12 @@ class OSCredentials(object):
   def __init__(self, options):
     self.environment_credentials()
     self.options_credentials(options)
+    self.credentials_available()
 
   def environment_credentials(self):
     try:
+      self.cred['s3_host']   = os.environ['s3_host']
```

Comment:
As discussed, a method for initializing a credential array is (IMO) not the right place to pass other data related to the checks

Reply:
I agree. Thanks